lib/diaspora_federation/entities/relayable.rb

```diff
@@ -179,41 +188,36 @@ def signature_order
 
       # @return [String] signature data string
       def signature_data
-        data = normalized_properties.merge(additional_xml_elements)
+        data = normalized_properties.merge(additional_data)
         signature_order.map {|name| data[name] }.join(";")
       end
 
-      # Override class methods from {Entity} to parse the xml
-      module ParseXML
-        private
-
-        # @param [Nokogiri::XML::Element] root_node xml nodes
-        # @return [Entity] instance
-        def populate_entity(root_node)
-          # Use all known properties to build the Entity (entity_data). All additional xml elements
-          # are respected and attached to a hash as string (additional_xml_elements). It also remembers
-          # the order of the xml-nodes (xml_order). This is needed to support receiving objects from
-          # the future versions of diaspora*, where new elements may have been added.
-          entity_data = {}
-          additional_xml_elements = {}
-
-          xml_order = root_node.element_children.map do |child|
-            xml_name = child.name
-            property = find_property_for_xml_name(xml_name)
-
-            if property
-              entity_data[property] = parse_element_from_node(xml_name, class_props[property], root_node)
-              property
-            else
-              additional_xml_elements[xml_name] = child.text
-              xml_name
-            end
+      # Override class methods from {Entity} to parse serialized data
+      module Parsing
+        # Does the same job as Entity.from_hash except of the following differences:
+        # 1) unknown properties from the properties_hash are stored to additional_data of the relayable instance
+        # 2) parent entity fetch is attempted
+        # 3) signatures verification is performed; property_order is used as the order in which properties are composed
+        # to compute signatures
+        # 4) unknown properties' keys must be of String type
+        #
+        # @see Entity.from_hash
+        def from_hash(properties_hash, property_order)
+          return if properties_hash.nil?
+          # Use all known properties to build the Entity (entity_data). All additional elements
+          # are respected and attached to a hash as string (additional_data). This is needed
+          # to support receiving objects from the future versions of diaspora*, where new elements may have been added.
+          entity_data = properties_hash
```

Comment:
I didn't notice this last time, but I think we don't need this line, when we don't change `properties_hash`. (See comment below)